server/src/main/kotlin/core/gameflow/showdown.kt

```diff
@@ -4,3 +4,42 @@ sealed class ShowdownAction
 
 data class ShowCards(val playerId: Int) : ShowdownAction()
 data class MuckCards(val playerId: Int) : ShowdownAction()
+
+fun resolveShowdown(handState: HandState): List<ShowdownAction> {
+
+    val orderedPlayers = when {
+        handState.lastAggressor == null -> handState.orderedPlayers(handState.buttonPosition + 1)
+        else -> handState.orderedPlayers(handState.lastAggressor.position)
+    }.filter { it.isInGame }
+
+    // when only one player is left in the game
+    if (orderedPlayers.size == 1)
+        return listOf(MuckCards(orderedPlayers.first().id))
+
+    // all players must show their cards if there are more streets to be dealt
+    if (handState.bettingRound != BettingRound.RIVER)
+        return orderedPlayers.map { ShowCards(it.id) }
+
+    // one player definitely wins with another if he has a better hand
+    // and he has put at least the same amount of chips into the pot
+    fun definitelyWins(player: Player, other: Player): Boolean {
+        val communityCards = handState.communityCards
+        return (player.hand(communityCards) > other.hand(communityCards)) and (player.chipsInPot >= other.chipsInPot)
+    }
+
+    val showingPlayers = mutableListOf<Player>()
+
+    orderedPlayers.forEach { player ->
+
+        // player should show the cards if no other player definitely wins with him
```

Comment:
it should read: '...no other showing player...'. Also I would add to the comment that last aggressor always shows cards, which is implicitly stated by code.

Reply:
Agreed about the first point. I don't see how the second suggestion would improve the code readability though :/

Reply:
I mean I would add that this loop continuously adds players to the showing pool and compares them in a sort of iterative process, because it wasn't clear for me from the start

Reply:
Ok, good point